### ISSUE TITLE
GPG status in grid

### ISSUE THREAD

**Issue by vbjay** [gitextensions/gitextensions#5528]
**Do you want to request a *feature* or report a *bug*?**
feature
**What is the current behavior?**
Showing the gpg tab is great but you have to go to the tab after selecting a commit. 
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
To look at a commit on the grid and know if it is a good signature, an invalid signature or what.  

1. A column that has a key with different colors and a tooltip or a way to see a window what the different key colors and what they mean.   (Easier done)
1. Different colors of grid row (harder because we already have logic around what the background color should be)

 The gpg tab is good to see the actual signature info.  
**Environment you encounter the issue:**

- GitExtensions version:master
- GIT version: NA
- OS version: NA
- .NET version: NA

**Did this work in previous version of GitExtensions (which)?**
no

**Comment by drewnoakes** [comment]
IIRC I looked at this an adding signature info to the `git log` call crippled performance. If we add this, I guess it'll need to populate lazily as it did for the multiline message indicator until relatively recently.

**Comment by vbjay** [comment]
Really?  Adding ```%G?``` killed performance?

**Comment by vbjay** [comment]
WOW!
FROM upstream/master
$ time git --no-pager log --pretty='%h %G?' >/dev/null

real    0m8.102s
user    0m0.015s
sys     0m0.031s

**Comment by vbjay** [comment]
Even worse.

$ git --no-pager log --all --pretty='%h %G?'  |wc -l
12100
time git --no-pager log --all --pretty='%h %G?' >/dev/null

real    0m37.509s
user    0m0.015s
sys     0m0.015s

**Comment by vbjay** [comment]
Yes.  This would need to be loaded in the background.  So the idea would be 

1. Run the normal log command to get the grid loaded.  
2. Spin off a background updater that updates the current commits with the status.
3.  Update the column that contains the gpg status.  

Could even optimize it to only update  the visible commits and a window more so scrolling a little bit shows the status.

**Comment by drewnoakes** [comment]
That's similar to how the multiline indicator worked.

Thinking about this again, it would probably be nicer overall to just run a single second log command in the background for all commits, rather than one per visible commit, as was done.

**Comment by vbjay** [comment]
I've thought of this more and more.  I think we should have a git log call that gets the grid loaded and an async load other data like body and such on a second run.  It could be changed to get body and gpg status and could even be limited to visible rows or visible rows +one screen up and down.  A post process commits.  I like the loading icon and then the loading status label in the corner like it is now.  We could even have it set so if the column is not visible, don't include it's argument in the format of the second log call. A column could have a format log property that would add arguments to the second log call.

**Comment by gpongelli** [comment]
GPG verification command was created as async task.
Initially I've created it as two icon on the right of Commit Header, but after discussion was decided to move it to another tab.

I don't know how the entire code is changed after v 2.50 .
